nsgcli/nsggrok_main.py

```diff
@@ -0,0 +1,155 @@
+
+    def do_q(self, _):
+        """Quits the program."""
+        self.do_quit()
+
+    @staticmethod
+    def do_quit():
+        """Quits the program."""
+        print('Quitting.')
+        raise SystemExit
+
+    def is_error(self, response):
+        if isinstance(response, dict) and 'error' in response:
+            error = response.get('error', '')
+            return error
+        if isinstance(response, list):
+            return self.is_error(response[0])
+        if isinstance(response, str):
+            resp_obj = json.loads(response)
+            return self.is_error(resp_obj)
+        return None
+
+    def summary(self):
+        print()
+        print('Base url: {0}'.format(self.base_url))
+        print('To exit, enter "quit" or "q" at the prompt')
+
+    ##########################################################################################
+    def do_log(self, cmd_args):
+        """
+        Parse syslog message with grok patterns
+
+        log <message>
+
+        Parameters:
+            message - Syslog Message to be parsed, read stdin if not specified
+
+        Examples:
+            log "<13>May 18 11:22:43 carrier sshd: SSHD_LOGIN_FAILED: Login failed for user 'root' from host '10.1.1.1'"
+            log <<< "<13>May 18 11:22:43 carrier sshd: SSHD_LOGIN_FAILED: Login failed for user 'root' from host 10.1.1.1"
+        """
+
+        if cmd_args is None:
+            self.read_stdin("log", self.pattern)
+        else:
+            self.call_grok_api("log", cmd_args, self.pattern)
+
+    ##########################################################################################
+    def do_text(self, cmd_args):
+        """
+        Parse text line with grok patterns.
+        Require 'pattern' parameter.
+
+        --pattern="Hello world of %{WORD:world_name}" text <text>
+
+        Parameters:
+            text - Text to be parsed, read stdin if not specified
+
+        Examples:
+            --pattern="Hello world of %{WORD:world_name}" text "Hello world of Grok"
+            --pattern="Hello world of %{WORD:world_name}" text <<< "Hello world of Grok"
+        """
+
+        if self.pattern is None:
+            print("'pattern' parameter is missing")
+            raise InvalidArgsException
+
+        if cmd_args is None:
+            self.read_stdin("", self.pattern)
+        else:
+            self.call_grok_api("", cmd_args, self.pattern)
+
+    def get_error(self, response):
+        """
+        if the response is in standard form (a dictionary with key 'error' or 'success') then
+        this function finds and returns the value of the key 'error'. Otherwise it returns
+        the whole response as a string
+        """
+        if isinstance(response, dict):
+            return response.get('error', str(response))
+        else:
+            return str(response)
+
+    def call_grok_api(self, parser, txt, pattern):
+        request = 'v2/grok/net/{0}/parser/{1}'.format(self.netid, parser)
```

Comment:
I believe the value of the parameter `pattern` is always `self.pattern`, if so, then function `call_grok_api()` does not really need this parameter

Reply:
right, simplified.